File: languages/go/monkey/ast/expression_statement.go
package ast

import "monkey/token"

type ExpressionStatement struct {
	Token      token.Token // the first token of the expression
	Expression Expression
}

func (es *ExpressionStatement) statementNode() {
}

func (es *ExpressionStatement) TokenLiteral() string {
	return es.Token.Literal
}

func (es *ExpressionStatement) String() string {
	if es.Expression == nil {
		return ""
	}

	return es.Expression.String()
}

File: languages/go/liner/src/Ast/ExpressionStatement.php
<?php

declare(strict_types=1);

namespace Serhii\Liner\Ast;

use Serhii\Liner\Token\Token;

readonly class ExpressionStatement implements Statement
{
    public function __construct(
        public Token $token,
        public ?Expression $expression,
    ) {
    }

    public function statementNode()
    {
    }

    public function tokenLiteral(): string
    {
        return $this->token->literal;
    }

    public function string(): string
    {
        if ($this->expression === null) {
            return '';
        }

        return $this->expression->string();
    }
}